File: app/src/main/java/com/kee0kai/thekey/room/KeyDatabase.java
package com.kee0kai.thekey.room;

import static com.kee0kai.thekey.App.DI;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

import com.kee0kai.thekey.room.dao.SettingDao;
import com.kee0kai.thekey.room.dao.StorageFilesDao;
import com.kee0kai.thekey.room.model.SettingPairEntry;
import com.kee0kai.thekey.room.model.StorageFileEntry;

@Database(entities =
        {
                StorageFileEntry.class,
                SettingPairEntry.class
        }, version = 1)
public abstract class KeyDatabase extends RoomDatabase {

    public abstract StorageFilesDao cachedFilesDao();

    public abstract SettingDao settingsDao();

    public static KeyDatabase create() {
        return Room.databaseBuilder(DI.app().application(), KeyDatabase.class, "db")
                .allowMainThreadQueries()
                .build();
    }


}

File: app_mobile/src/main/kotlin/com/github/klee0kai/thekey/app/data/room/KeyDatabase.kt
package com.github.klee0kai.thekey.app.data.room

import androidx.room.Database
import androidx.room.Room.databaseBuilder
import androidx.room.RoomDatabase
import com.github.klee0kai.thekey.app.data.room.dao.ColorGroupDao
import com.github.klee0kai.thekey.app.data.room.dao.SettingDao
import com.github.klee0kai.thekey.app.data.room.dao.StorageFilesDao
import com.github.klee0kai.thekey.app.data.room.entry.ColorGroupEntry
import com.github.klee0kai.thekey.app.data.room.entry.SettingPairEntry
import com.github.klee0kai.thekey.app.data.room.entry.StorageFileEntry
import com.github.klee0kai.thekey.app.di.DI

@Database(
    entities = [
        StorageFileEntry::class,
        ColorGroupEntry::class,
        SettingPairEntry::class,
    ],
    version = 1,
)
abstract class KeyDatabase : RoomDatabase() {

    abstract fun storagesDao(): StorageFilesDao

    abstract fun colorGroupsDao(): ColorGroupDao

    abstract fun settingsDao(): SettingDao


    companion object {
        fun create(): KeyDatabase =
            databaseBuilder(DI.ctx(), KeyDatabase::class.java, "db")
                .allowMainThreadQueries()
                .build()
    }

}